src/lamplib/tasks/toolchain.py

```diff
@@ -52,10 +51,11 @@ def fetch_and_install_toolchain(url, install_dir):
             logging.info('Finished Downloading gennytoolchain as %s', toolchain_tarball)
 
         logging.info('Extracting gennytoolchain into %s, please wait...', toolchain_dir)
-        tarball = tarfile.open(toolchain_tarball)
 
         shutil.rmtree(toolchain_dir, ignore_errors=True)
-        tarball.extractall(toolchain_dir)
+        os.mkdir(toolchain_dir)
+        # use tar(1) because python's TarFile was inexplicably truncating the tarball
+        subprocess.run(['tar', '-xzf', toolchain_tarball, '-C', toolchain_dir])
```

Comment:
Is this doing any kind of error handling? It looks like the `check` parameter defaults to false. https://docs.python.org/3/library/subprocess.html#subprocess.run

Reply:
👍 added `check=True`.